base/loading.jl

```diff
@@ -415,14 +415,15 @@ end
 evalfile(path::AbstractString, args::Vector) = evalfile(path, UTF8String[args...])
 
 function create_expr_cache(input::AbstractString, output::AbstractString)
-    isfile(output) && rm(output)
```

Comment:
Deleting first seems cleaner. This avoids the following scenario:
- Build 1 is started, fails, but takes a long time
- Build 2 starts later, succeeds, and produces a valid cache file
- Build 1 finally finishes, and deletes the newly created cache file

Deleting first ensures that the build that starts last either produces a cache file, or deletes it.